### PR TITLE
Add `any` function to the function module

### LINKED ISSUE #593: [FEATURE REQUEST] Support for ANY/SOME array function/comparison
Hi, I have some use cases where it's needed to check if some string in present on a array. For those cases, I found myself wanting to use the ["= ANY()"](https://www.postgresql.org/docs/current/functions-comparisons.html#id-1.5.8.30.16) construct on my Postgres queries. 

Here is an example of such:
```ts
let query = this.db
            .selectFrom('notifications')
            .leftJoin('readReceipts', 'notifications.id', 'readReceipts.notificationId')
            .select((eb) => [
                'id',
                'oneSignalId',
                'heading',
                'body',
                'data',
                'notifications.createdAt',
                'type',
                eb('notificationId', 'is not', null).as('read'),
            ])
            .where(sql`${username}`, '=', sql`ANY(destination_usernames)`)

if (lastId) {
    query = query.where('id', '<', lastId)
}
if (take) {
    query = query.limit(take)
}

const notifications = await query.orderBy('id', 'desc').execute()

console.log(notifications[0])

return notifications
```

A playground with it and the DB types (mostly): [Playground Link](https://kyse.link/?p=b&i=haFkqHBvc3RncmVzoXamMC4yNi4woXPaBNFpbXBvcnQgeyBHZW5lcmF0ZWQsIENvbHVtblR5cGUgfSBmcm9tICdreXNlbHknCgpleHBvcnQgdHlwZSBUaW1lc3RhbXAgPSBDb2x1bW5UeXBlPERhdGUsIERhdGUgfCBzdHJpbmcsIERhdGUgfCBzdHJpbmc-CgpkZWNsYXJlIGdsb2JhbCB7CiAgaW50ZXJmYWNlIERCIHsKICAgIGFkdmlzb3JOb3RpZmljYXRpb25QcmVmZXJlbmNlczogQWR2aXNvck5vdGlmaWNhdGlvblByZWZlcmVuY2VzCiAgICBkZWZhdWx0Tm90aWZpY2F0aW9uUHJlZmVyZW5jZXM6IERlZmF1bHROb3RpZmljYXRpb25QcmVmZXJlbmNlcwogICAgbm90aWZpY2F0aW9uczogTm90aWZpY2F0aW9ucwogICAgcmVhZFJlY2VpcHRzOiBSZWFkUmVjZWlwdHMKICB9CgogIGludGVyZmFjZSBBZHZpc29yTm90aWZpY2F0aW9uUHJlZmVyZW5jZXMgewogICAgdXNlcm5hbWU6IHN0cmluZwogICAgbm90aWZpY2F0aW9uVHlwZTogJ2FwcG9pbnRtZW50JyB8ICdiaXJ0aGRheScgfCAnbnBzJyB8ICd0ZWQnCiAgICBlbmFibGVkOiBib29sZWFuCiAgICBleHRyYVBhcmFtZXRlcnM6IFJlY29yZDxzdHJpbmcsIHN0cmluZz4gfCBudWxsCn0KCmludGVyZmFjZSBEZWZhdWx0Tm90aWZpY2F0aW9uUHJlZmVyZW5jZXMgewogICAgbm90aWZpY2F0aW9uVHlwZTogJ2FwcG9pbnRtZW50JyB8ICdiaXJ0aGRheScgfCAnbnBzJyB8ICd0ZWQnCiAgICBlbmFibGVkOiBHZW5lcmF0ZWQ8Ym9vbGVhbj4KICAgIGV4dHJhUGFyYW1ldGVyczogUmVjb3JkPHN0cmluZywgc3RyaW5nPiB8IG51bGwKfQoKaW50ZXJmYWNlIE5vdGlmaWNhdGlvbnMgewogICAgb25lU2lnbmFsSWQ6IHN0cmluZyB8IG51bGwKICAgIGNyZWF0ZWRBdDogR2VuZXJhdGVkPFRpbWVzdGFtcD4KICAgIHR5cGU6ICdhcHBvaW50bWVudCcgfCAnYmlydGhkYXknIHwgJ25wcycgfCAndGVkJwogICAgZGF0YTogUmVjb3JkPHN0cmluZywgc3RyaW5nPgogICAgYm9keTogc3RyaW5nCiAgICBoZWFkaW5nOiBzdHJpbmcgfCBudWxsCiAgICBpZDogR2VuZXJhdGVkPHN0cmluZz4KICAgIHNlbmRpbmdFcnJvcjogc3RyaW5nIHwgbnVsbAogICAgZGVzdGluYXRpb25Vc2VybmFtZXM6IHN0cmluZ1tdCiAgICBwdXNoUGF5bG9hZDogUmVjb3JkPHN0cmluZywgc3RyaW5nPiB8IG51bGwKfQoKaW50ZXJmYWNlIFJlYWRSZWNlaXB0cyB7CiAgICBub3RpZmljYXRpb25JZDogc3RyaW5nCiAgICByZWFkQnk6IHN0cmluZwogICAgY3JlYXRlZEF0OiBHZW5lcmF0ZWQ8VGltZXN0YW1wPgp9Cn2hcdoDHGltcG9ydCB7IHNxbCB9IGZyb20gImt5c2VseSIKCmxldCBsYXN0SWQgPSAiYXNkYXNkc2FkIgpsZXQgdGFrZSA9IDUKbGV0IHVzZXJuYW1lID0gInNvbWVfdXNlciIKCmxldCBxdWVyeSA9IGt5c2VseQogICAgICAgICAgICAuc2VsZWN0RnJvbSgnbm90aWZpY2F0aW9ucycpCiAgICAgICAgICAgIC5sZWZ0Sm9pbigncmVhZFJlY2VpcHRzJywgJ25vdGlmaWNhdGlvbnMuaWQnLCAncmVhZFJlY2VpcHRzLm5vdGlmaWNhdGlvbklkJykKICAgICAgICAgICAgLnNlbGVjdCgoZWIpID0-IFsKICAgICAgICAgICAgICAgICdpZCcsCiAgICAgICAgICAgICAgICAnb25lU2lnbmFsSWQnLAogICAgICAgICAgICAgICAgJ2hlYWRpbmcnLAogICAgICAgICAgICAgICAgJ2JvZHknLAogICAgICAgICAgICAgICAgJ2RhdGEnLAogICAgICAgICAgICAgICAgJ25vdGlmaWNhdGlvbnMuY3JlYXRlZEF0JywKICAgICAgICAgICAgICAgICd0eXBlJywKICAgICAgICAgICAgICAgIGViKCdub3RpZmljYXRpb25JZCcsICdpcyBub3QnLCBudWxsKS5hcygncmVhZCcpLAogICAgICAgICAgICBdKQogICAgICAgICAgICAud2hlcmUoc3FsYCR7dXNlcm5hbWV9YCwgJz0nLCBzcWxgQU5ZKGRlc3RpbmF0aW9uX3VzZXJuYW1lcylgKQoKaWYgKGxhc3RJZCkgewogICAgcXVlcnkgPSBxdWVyeS53aGVyZSgnaWQnLCAnPCcsIGxhc3RJZCkKfQppZiAodGFrZSkgewogICAgcXVlcnkgPSBxdWVyeS5saW1pdCh0YWtlKQp9Cgpjb25zdCBub3RpZmljYXRpb25zID0gYXdhaXQgcXVlcnkub3JkZXJCeSgnaWQnLCAnZGVzYycpLmV4ZWN1dGUoKTuhY8M=)

### PR DESCRIPTION
**Pull request by koskimas**
Closes #593

### CODE REVIEW

File: test/node/src/expression.test.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ for (const dialect of DIALECTS) {
               or([
                 not(eb('first_name', '=', 'Jennifer')),
                 eb(eb('id', '+', 1), '>', 10),
-                eb(ref('id'), 'in', val([10, 20, 30])),
+                eb(ref('id'), 'in', [10, 20, 30].map(val)),
```
Comment (igalklebanov) [inline test/node/src/expression.test.ts:41]: let's not forget to mention the breaking change here.

**Review by igalklebanov — DISMISSED** [review]
LGTM! 🚀

**Review by igalklebanov — COMMENTED** [review]
LGMT! 🚀 

Missing some [MySQL](https://dev.mysql.com/doc/refman/8.0/en/any-in-some-subqueries.html) tests.

**Comment by koskimas** [comment]
MySQL doesn't have an array type, right?

**Comment by koskimas** [comment]
> Oh crap. The MySQL syntax probably won't work on Postgres.

It did work 😅. @igalklebanov I've added a test for a subquery `any`. Nice catch!

File: test/node/src/expression.test.ts (hunk 1 above)
Comment (koskimas) [reply test/node/src/expression.test.ts:41]: It's highly unlikely anyone has used that. I'd also argue that was a bugfix rather than a breaking change.

**Review by igalklebanov — APPROVED** [review]
Let's go! 🚀

**Comment by maccman** [comment]
@koskimas @igalklebanov I can't get this working on `Generated` columns.

CleanShot 2023-08-26 at 09 19 01@2x

Works on a straight `string[]` column though.

**Pull request merged**